relay-config/src/config.rs

```diff
@@ -643,6 +700,160 @@ impl Config {
         Ok(config)
     }
 
+    /// Override configuration with values coming from other sources (e.g. env variables or
+    /// command line parameters)
+    pub fn override_config(
+        &mut self,
+        overrides: OverridableConfig,
+    ) -> Result<&mut Self, ConfigError> {
+        let relay = &mut self.values.relay;
+
+        if let Some(upstream) = overrides.upstream {
+            relay.upstream = UpstreamDescriptor::from_str(&upstream).map_err(|err| {
```

Comment:
Instead of importing `FromStr`, it would be more idiomatic to call parse here. The type can be inferred by Rust.

```rust
relay.upstream = upstream.parse().map_err(...);
```

Reply:
cool